### ISSUE TITLE
opt: add more exhaustive partial index logic tests

### ISSUE THREAD

**Issue by mgartner** [cockroachdb/cockroach#51528]
We should add logic tests that randomize families to catch some of these bugs.

Example:

```
create table abc(a int, b int, c int, index (c) where b>1, family(a), family(b), family(c));

insert into abc values (1, 2, 3);

update abc set b=b+1 where a=1;
*
* ERROR: [n1,client=127.0.0.1:61904,hostssl,user=root] a SQL panic has occurred while executing the following statement:
* UPDATE abc SET b = b + 1 WHERE a = 1
*
*
* ERROR: [n1,client=127.0.0.1:61904,hostssl,user=root] a panic has occurred!
*
panic: runtime error: index out of range [2] with length 2 [recovered]
        panic: runtime error: index out of range [2] with length 2
```

**Comment by jordanlewis** [comment]
All logic tests already randomize their column families during `CREATE TABLE` statements, unless families are explicitly specified.

**Comment by mgartner** [comment]
The panic above was fixed by #51618.

Leaving this issue open for now because we still need more logic tests for partial indexes that cover a smattering of mutations and selects.
